### PR TITLE
Open Synthetics recorder directly from kibana via link

### PR DESCRIPTION
**Pull request by shahzad31**
> ⚠️ As requested by @shahzad31, I (@lucasfcosta) am now responsible for this PR, so I'll be the ones pushing commits and responding to comments in this issue.

# Summary

This PR adds the ability to open Recorder directly from Kibana (or any other website) through a link using the `elastic-synthetics-recorder-protocol`.

You could, for example, use the following `EuiLink` to open the recorder:

```
<EuiLink href={`elastic-synthetics-recorder://`}>Open Script Recorder</EuiLink>
```

> Please notice that this PR/task does not include adding the relevant links in Kibana, and that that will be covered separately in https://github.com/elastic/kibana/issues/120638.

For now, there is no way to determine that the recorder is installed or not. Therefore, users which do not have the recorder installed will see the following standard error:

Standard error when the script is not installed

Below is a gif of the link working:

# How to test this PR

1. Clone this branch and package the application using `npm run package`
2. Once the application is packaged, mount its `dmg` and install the `.app` into `Applications`.
3. Create a new folder `example-folder` and add the following HTML file as `index.html` to it:
    ```
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <title>Test Recorder Link</title>
    </head>
    <body>
      <a href="elastic-synthetics-recorder://">Open recorder</a>
    </body>
    </html>
    ```
4. Serve the HTML file in that folder using `npx http-server example-folder`
5. Access the `localhost` URL in which you're serving your file and click the `Open recorder` link. The script recorder should open.

### CODE REVIEW

**Comment by lucasfcosta** [comment]
As discussed with @shahzad31 on Slack, I'll be picking this issue up as he's busy at the moment and I'll be working on the recorder more often.

**Review by vigneshshanmugam — APPROVED** [review]

**Review by justinkambic — APPROVED** [review]
I've looked into it a bit further and haven't found an OS-specific reason we can't merge this, so LGTM.

**Comment by lucasfcosta** [comment]
As mentioned on slack by @justinkambic we'll wait until we have heard back from the security team on whether this can impact notarization:

> One thing I’m wondering about this, is will it impact our desire to notarize the app via Elastic’s Apple Dev account. I haven’t fully investigated this yet, it seems innocuous and a really minimal change, but it might be good to verify before we merge this.

**Comment by shahzad31** [comment]
> As mentioned on slack by @justinkambic we'll wait until we have heard back from the security team on whether this can impact notarization:
> 
> > One thing I’m wondering about this, is will it impact our desire to notarize the app via Elastic’s Apple Dev account. I haven’t fully investigated this yet, it seems innocuous and a really minimal change, but it might be good to verify before we merge this.

I can't possibly think why it will impact notarisation? Did you hear back @lucasfcosta , let's push this otherwise, then we can see how we want to use it in Kibana.

**Comment by lucasfcosta** [comment]
@shahzad31 I've pinged the security team about this subject this morning, and will open a consulting request on their repo as I was instructed. I'll keep this PR up-to-date with their responses and with the issue once it's there.

**Comment by lucasfcosta** [comment]
⭐  Update: This is now being tracked by the security team in their consulting repo, as per their request:

* Issue: https://github.com/elastic/infosec/issues/8813

**Comment by lucasfcosta** [comment]
Merging as per the security team's approval. Let's make sure we do another security review before passing data from Kibana to the recorder through the link.

**Pull request merged**
